### ISSUE TITLE
pipe.enable_attention_slicing() broken on main

### ISSUE THREAD

**Issue by rabidgremlin** [huggingface/diffusers#1405]
### Describe the bug

pipe.enable_attention_slicing() appears to be broken on main. Discovered when attempting to run SD 2.0 example code.

### Reproduction

Setup:
```
pip install --upgrade git+https://github.com/huggingface/diffusers.git transformers accelerate scipy ftfy
pip install -U --force-reinstall torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu117
```

Code:
```
from diffusers import StableDiffusionPipeline, EulerDiscreteScheduler
import torch

model_id = "stabilityai/stable-diffusion-2"

# Use the Euler scheduler here instead
scheduler = EulerDiscreteScheduler.from_pretrained(model_id, subfolder="scheduler")
pipe = StableDiffusionPipeline.from_pretrained(model_id, scheduler=scheduler, revision="fp16", torch_dtype=torch.float16)
pipe = pipe.to("cuda")

pipe.enable_attention_slicing()

prompt = "a photo of an astronaut riding a horse on mars"
image = pipe(prompt, height=768, width=768).images[0]
    
image.save("astronaut_rides_horse.png")
```

### Logs

```shell
(.venv) F:\sd20diffusersmain>python sd20test.py
Fetching 12 files: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 12/12 [00:00<00:00, 12072.83it/s] You have disabled the safety checker for <class 'diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion.StableDiffusionPipeline'> by passing `safety_checker=None`. Ensure that you abide to the conditions of the Stable Diffusion license and do not expose unfiltered results in services or applications open to the public. Both the diffusers team and Hugging Face strongly recommend to keep the safety filter enabled in all public facing circumstances, disabling it only for use-cases that involve analyzing network behavior or auditing its results. For more information, please have a look at https://github.com/huggingface/diffusers/pull/254 .Traceback (most recent call last):
  File "sd20test.py", line 11, in <module>    pipe.enable_attention_slicing()  File "F:\sd20diffusersmain\.venv\lib\site-packages\diffusers\pipelines\stable_diffusion\pipeline_stable_diffusion.py", line 170, in enable_attention_slicing
    slice_size = self.unet.config.attention_head_dim // 2
TypeError: unsupported operand type(s) for //: 'list' and 'int'
```
```


### System Info

- `diffusers` version: 0.9.0.dev0
- Platform: Windows-10-10.0.22621-SP0
- Python version: 3.8.10
- PyTorch version (GPU?): 1.13.0+cu117 (True)
- Huggingface_hub version: 0.11.0
- Transformers version: 4.24.0
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

**Comment by rabidgremlin** [comment]
Related issue #1398

**Comment by kashif** [comment]
Thank you @rabidgremlin for the report. I believe PR  #1397 is the fix for this.

**Comment by pcuenca** [comment]
Fixed by #1397 as @kashif said.

**Comment by rabidgremlin** [comment]
Great. Have tested (by doing a `pip install --upgrade --force-reinstall  git+https://github.com/huggingface/diffusers.git` and can confirm that it is now working! Thanks
